### PR TITLE
Add rule details view with run history

### LINKED ISSUE #122: Add run history for rules

### PR DESCRIPTION
**Pull request by thu5**
This PR adds the run details UI which includes details about the rule and the run history. Resolves #122 and #77.

success  failure

### CODE REVIEW

File: lib/src/rule_component.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<div class="mb-3" *ngIf="view == View.listView">
+    <div class="flex-column">
+        <rule-list (detailedRule)="changeDetailedRule($event)" class="mb-5"></rule-list>
+        <button type="button" (click)="toCreateView()" class="mt-3 btn btn-success">Create Rule</button>
+    </div>
+</div>
+<div *ngIf="view == View.createView">
+    <rule-creator></rule-creator>
+    <button type="button" (click)="toListView()" class="btn btn-secondary">Back</button>
+</div>
```
Comment (gg-qinp) [inline lib/src/rule_component.html:10]: does Git have a delta diff view which omits the diff from previous RP?

File: lib/src/service/rule_service.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      'Authorization': 'Bearer $idToken'
+    };
+    final response = await get(rulesEndpoint, headers: headers);
+    return GetRulesResponse.fromBuffer(response.bodyBytes).rules;
+  }
+
+  Future<GetRunHistoryResponse> getRunHistory(
+      String idToken, String ruleId) async {
+    final headers = {
+      'Content-Type': 'application/x-protobuf',
```
Comment (gg-qinp) [inline lib/src/service/rule_service.dart]: consider make this a static function instead of defining it repeatedly.

**Review by gg-qinp — APPROVED** [review]

File: lib/src/service/rule_service.dart (hunk 2 above)
Comment (thu5) [reply lib/src/service/rule_service.dart]: Ok, sure

File: lib/src/rule_component.html (hunk 1 above)
Comment (thu5) [reply lib/src/rule_component.html:10]: I don't think so :/

**Review by cathyang-work — APPROVED** [review]

**Pull request merged**